Resources/Prototypes/Entities/Objects/Weapons/Melee/chainsaw.yml

```diff
@@ -44,6 +75,9 @@
         reagents:
         - ReagentId: WeldingFuel
           Quantity: 300
-        maxVol: 300
+  - type: Welder
+    fuelConsumption: 3
+    fuelLitCost: 2
+    tankSafe: true
   - type: UseDelay
```

Comment:
Please don't use weldercomponent for this, split it out into a reagent provider.

Reply:
See also https://github.com/space-wizards/space-station-14/issues/18689

Reply:
> See also #18689

Thanks. Then I should deep dive a little into ToggleSystem. Will try to make required component for cases like this.